source/ui/BioXAS/BioXASControlButton.cpp

```diff
@@ -0,0 +1,69 @@
+#include "BioXASControlButton.h"
+
+BioXASControlButton::BioXASControlButton(AMControl *control, QWidget *parent) :
+	AMToolButton(parent)
+{
+	// Initialize class variables.
+
+	control_ = 0;
+
+	greenValueSet_ = false;
+	greenValue_ = 0;
+
+	// Current settings.
+
+	setControl(control);
+
+	refresh();
```

Comment:
we called `refresh()` in `setControl`. I think we don't need to call this again.